src/distilabel/utils/serialization.py

```diff
@@ -0,0 +1,154 @@
+
+import importlib
+import json
+import os
+from dataclasses import asdict
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Dict, Generic, Optional, TypeVar
+
+if TYPE_CHECKING:
+    from distilabel.tasks.base import Task
+
+
+T = TypeVar("T")
+
+TASK_FILE_NAME = "task-distilabel.json"
+
+
+def load_from_dict(template: Dict[str, Any]) -> Generic[T]:
+    """Reads a template (a class serialized) and returns a the instance
+    contained.
+
+    Args:
+        template (Dict[str, Any]): Dict containing the template, the dict serialized.
+
+    Returns:
+        Generic[T]: Instance contained in the template
+    """
+    type_info = template.pop("__type_info__")
+    mod = importlib.import_module(type_info["module"])
+    cls = getattr(mod, type_info["name"])
+    instance = cls(**template)
+    return instance
+
+
+def load_task_from_disk(path: Path) -> "Task":
+    """Loads a task from disk.
+
+    Args:
+        path: The path to the task.
+
+    Returns:
+        Task: The task.
+    """
+    task_path = path / TASK_FILE_NAME
+    if not task_path.exists():
+        raise FileNotFoundError(f"The task file does not exist: {task_path}")
+
+    task_content = read_json(task_path)
+    task = load_from_dict(task_content)
+    return task
+
+
+def write_json(filename: Path, data: Dict[str, Any]) -> None:
+    """Writes a json file to the given path, creates the parent dir.
+
+    Args:
+        filename (Path): Name of the file.
+        data (Dict[str, Any]): Dict to be written as json.
+    """
+    filename.parent.mkdir(parents=True, exist_ok=True)
+    with open(filename, "w") as f:
+        json.dump(data, f, indent=2)
+
+
+def read_json(filename: Path) -> Dict[str, Any]:
+    """Read a json file from disk.
+
+    Args:
+        filename (Path): Name of the json file.
+
+    Returns:
+        Dict[str, Any]: Dict containing the json data.
+    """
+    with open(filename, "r") as file:
+        return json.load(file)
+
+
+class _Serializable:
+    """Base class for serializable classes.
+    It provides the means to serialize and deserialize.
+
+    We currently defien the tasks as dataclasses, in which case we can
+    use the `asdict` method to serialize the class, but we may need to review
+    this if we decide to remove the dataclasses.
+    Other than the default content, the obtained from `dataclasses.asdict`, we
+    we store create a __type_info__ variable to store the relevant information
+    to load back the class.
+    """
+
+    __type_info__: Dict[str, Any] = {}
+
+    def dump(self) -> Dict[str, Any]:
+        """Transforms the class into a dict to write to a file.
+
+        Returns:
+            Dict[str, Any]: Serializable content of the class.
+        """
+        _dict = asdict(self)
+        _dict["__type_info__"] = {
+            "module": type(self).__module__,
+            "name": type(self).__name__,
+        }
+        return _dict
+
+    def save(self, path: Optional[os.PathLike] = None) -> None:
```

Comment:
should we add some checks to llow creating the path if not present?

Reply:
there are checks in place, isn't it?
```python
    def save(self, path: Optional[os.PathLike] = None) -> None:
        ...
        if path is None:
            path = Path.cwd() / TASK_FILE_NAME
        else:
            path = Path(path)
            if path.suffix == "":
                # If the path has no suffix, assume the user just wants a folder to write the task
                path = path / TASK_FILE_NAME
```

Reply:
I mean if the path does not exists but probably we'll also see that during the writing